docs/external-node.md

```diff
@@ -0,0 +1,532 @@
+
+- To apply Antrea NetworkPolicy to an external Node.
+- You want the same security configurations on the external Node for all
+  Operating Systems.
+
+This guide demonstrates how to configure `ExternalNode` to achieve the above
+result.
+
+# Prerequisites
+
+`ExternalNode` is introduced in v1.8 as an alpha feature. A feature gate
+`ExternalNode` must be enabled on the `antrea-controller` and `antrea-agent`in
+the configuration. The configuration for `antrea-controller` is modified in the
+`antrea-config` ConfigMap as follows for the feature to work:
+
+```yaml
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: antrea-config-dcfb6k2hkm
+  namespace: kube-system
+data:
+  antrea-controller.conf: |
+    featureGates:
+      ExternalNode: true
+```
+
+Service `antrea` is created for `antrea-controller` to accept connections from
+`antrea-agent`, and the type is `ClusterIP` by default. Since an external Node
+is running outside the Kubernetes cluster, the `ClusterIP` address is not
+accessible. So a `NodePort` or `LoadBalancer` typed Service for `antrea` is
+required.
+
+Since `antrea-agent` is running on an external Node which is not managed by
+Kubernetes, a single configuration file is required to locate on each machine
+where `antrea-agent` is running. 
+
+For the detail configuration options when running `antrea-agent`on an external
+Node, please refer to the [sample configuration file](../build/yamls/externalnode/conf/antrea-agent.conf).
+
+# The ExternalNode resource
+
+A typical `ExternalNode` resource example:
+
+```yaml
+apiVersion: crd.antrea.io/v1alpha1
+kind: ExternalNode
+metadata:
+  name: vm1
+  namespace: vm-ns
+  labels:
+    role: db
+spec:
+  interfaces:
+    - ips: [ "172.16.100.3" ]
+      name: ""
+```
+
+## Name and Namespace
+
+The `name` field in an `ExternalNode` is unique, which is the key to tell
+`antrea-agent` where it is running on. It is also used as the key in the
+connection from `antrea-agent` to `antrea-controller`. The `ExternalNode` name
+is given to `antrea-agent` via an environment variable `NODE_NAME`, and
+`antrea-agent` will use the hostname to find the `ExternalNode` resource if it
+is not set.
+
+`ExternalNode` resource is working in `namespace` scope. The `namespace` is
+given to `antrea-agent` with option `externalNodeNamespace` in [antrea-agent.conf](../build/yamls/externalnode/conf/antrea-agent.conf).
+
+```yaml
+externalNodeNamespace: vm-ns
+```
+
+## Interfaces
+
+The `interfaces` field specifies the list of the network interfaces expected to
+be guarded by Antrea NetworkPolicy. One interface at least is required. The
+`name` or `ips` can be used to identify the target interface. **The field `ips`
+must be provided in the CRD**, but `name` is optional. Multiple IPs on a single
+interface is supported.
+
+`antrea-controller` creates an `ExternalEntity` for each interface whenever it
+finds an `ExternalNode is created. The created `ExternalEntity` has the
```

Comment:
the markdown formatting is wrong here

suggestion:
> whenever an `ExternalNode` is created.

Reply:
updated.